table/tables/partition.go

```diff
@@ -166,6 +172,43 @@ func generatePartitionExpr(tblInfo *model.TableInfo) (*PartitionExpr, error) {
 	}, nil
 }
 
+func generateHashPartitionExpr(tblInfo *model.TableInfo) (*PartitionExpr, error) {
+	var column *expression.Column
+	// The caller should assure partition info is not nil.
+	pi := tblInfo.GetPartitionInfo()
+	ctx := mock.NewContext()
+	partitionPruneExprs := make([]expression.Expression, 0, len(pi.Definitions))
+	var buf bytes.Buffer
+	dbName := model.NewCIStr(ctx.GetSessionVars().CurrentDB)
```

Comment:
I'm not sure the mocked context have the right session vars.
Could you get the `dbName` from `tblInfo`?

Reply:
`ctx.GetSessionVars().CurrentDB` is `""`.  function `generatePartitionExpr` have the same problem too.
I can't get `dbName` from `tblInfo`.
Maybe it is a bug, I will fix in next PR, This PR is big enough.

Reply:
ok